source/tutorial/cpp.txt

```diff
@@ -0,0 +1,647 @@
+              _dbManager.addNew(
+                _homeControllerState.newTaskIsComplete, 
+                _homeControllerState.newTaskSummary, 
+                _homeControllerState.newTaskPriority);
+              _homeControllerState.newTaskSummary = "";
+              _homeControllerState.newTaskIsComplete = false;
+              _homeControllerState.newTaskPriority = HomeControllerState::DEFAULT_TASK_PRIORITY;
+            });
+
+            auto newTaskLayout = ftxui::Container::Horizontal(
+              {inputNewTaskSummary, newTaskCompletionStatus, newTaskPriorityDropdown, saveButton});
+
+      #. Finally, farther down in the ``home_controller.cpp`` file, add the
+         new UI element to the ``auto itemListRenderer``:
+
+         .. code-block:: cpp
+            :emphasize-lines: 3
+
+            inputNewTaskSummary->Render() | ftxui::flex,
+            newTaskCompletionStatus->Render() | ftxui::center,
+            newTaskPriorityDropdown->Render(),
+            saveButton->Render(),
+
+         This renders the new element right before the Save button in the UI.
+
+   .. step:: Save the New Property to the Database
+
+      A. In the ``managers`` directory, go to ``database_manager.hpp``. 
+         Update the ``addNew()`` function signature to include the
+         ``int newItemPriority`` we pass in from the ``home_controller.cpp``:
+
+         .. code-block:: cpp
+            :emphasize-lines: 4
+
+            void addNew(
+              bool newItemIsComplete,
+              std::string newItemSummary,
+              int newItemPriority);
+
+      #. Now go to ``database_manager.cpp``, and update the ``addNew()``
+         implementation. Add ``int newItemProperty`` to the function arguments:
+
+         .. code-block:: cpp
+            :emphasize-lines: 4
+
+            void DatabaseManager::addNew(
+              bool newItemIsComplete,
+              std::string newItemSummary,
+              int newItemPriority) {
+              ...implementation...
+              }
+         
+         Add a new line in the function to set the value of the ``priority`` 
+         property when we save the ``Item`` to the database:
+
+         .. code-block:: cpp
+
+            .priority = newItemPriority
+
+         Your ``addNew()`` implementation should now resemble:
+
+         .. code-block:: cpp
+            :emphasize-lines: 1, 6
+
+            void DatabaseManager::addNew(bool newItemIsComplete, std::string newItemSummary, int newItemPriority) {
+              auto item = realm::Item {
+                .isComplete = newItemIsComplete,
+                .summary = std::move(newItemSummary),
+                .owner_id = _userId,
+                .priority = newItemPriority
+              };
+
+              _database->write([&]{
+                _database->add(std::move(item));
+              });
+            }
+
+   .. step:: Run and Test
+    
+      At this point, you can run the application again. Log in using the account 
```

Comment:
I had to rebuild before running. not sure if that's not the intended workflow or if it's just an assumed step to "run the application", but it tripped me up for a second when I didn't see my changes

Reply:
It's a miss on my part - in Xcode/SwiftUI, running the app builds it with the latest changes. I forgot to update this for C++. I added the info here.